### PR TITLE
Added "smart" fraction and MathLatexBuilder

### PR DESCRIPTION
**Pull request by flokixdev**

### CODE REVIEW

File: src/Entity/Fraction.php [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+namespace Mathematicator\Numbers\Entity;
+
+use Brick\Math\BigNumber;
+use Stringable;
+
+/**
+ * Entity to store simple and compound fractions
+ */
+class Fraction
```
Comment (janbarasek) [inline src/Entity/Fraction.php:13]: Fraction should be subtype of `SmartNumber` which can represent fraction too.

File: src/Entity/Fraction.php [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	 * @var Fraction|null
+	 */
+	protected $parentInDenominator;
+
+
+	/**
+	 * @param string|Stringable|BigNumber|Fraction|null $numerator optional
+	 * @param string|Stringable|BigNumber|Fraction|null $denominator optional
+	 */
+	public function __construct($numerator = null, $denominator = null)
```
Comment (janbarasek) [inline src/Entity/Fraction.php:46]: Why the input type is not simple `string` or `SmartNumber` only?

Comment (flokixdev) [reply src/Entity/Fraction.php:46]: Because it is not a parser.

Comment (janbarasek) [reply src/Entity/Fraction.php:46]: What is the purpose of this service? Why can't you use tokenization?

Comment (flokixdev) [reply src/Entity/Fraction.php:46]: It is not a service, it's entity representation of a fraction that I want to use for calculator. Simple array is not enough (e.g. for traversing).

File: src/Entity/Fraction.php (hunk 1 above)
Comment (flokixdev) [reply src/Entity/Fraction.php:13]: I think that number should be number and fraction should be fraction, but SmartNumber can use it internally.

File: src/Entity/Fraction.php (hunk 2 above)
Comment (flokixdev) [reply src/Entity/Fraction.php:46]: Tokenizer can be used to create this entity.

Comment (janbarasek) [reply src/Entity/Fraction.php:46]: Ok, thanks. Now I understand.

**Comment by janbarasek** [comment]
Thanks!

**Pull request merged**